mmdet3d/models/backbones/dla.py

```diff
@@ -0,0 +1,340 @@
+                build_conv_layer(
+                    conv_cfg,
+                    in_channels,
+                    out_channels,
+                    1,
+                    stride=1,
+                    bias=False),
+                dla_build_norm_layer(norm_cfg, out_channels)[1])
+
+    def forward(self, x, residual=None, children=None):
+        children = [] if children is None else children
+        bottom = self.downsample(x) if self.downsample is not None else x
+        residual = self.project(bottom) if self.project is not None else bottom
+        if self.level_root:
+            children.append(bottom)
+        x1 = self.tree1(x, residual)
+        if self.levels == 1:
+            x2 = self.tree2(x1)
+            feat_list = [x2, x1] + children
+            x = self.root(feat_list)
+        else:
+            children.append(x1)
+            x = self.tree2(x1, children=children)
+        return x
+
+
+@BACKBONES.register_module()
+class DLANet(nn.Module):
+    """DLA backbone <https://arxiv.org/abs/1707.06484>`_.
+
+    Args:
+        depth (int): Depth of dla. Default: 34.
+        in_channels (int): Number of input image channels. Default: 3.
+        norm_cfg (dict): Dictionary to construct and config norm layer.
+                         Default: None.
+        conv_cfg (dict): Dictionary to construct and config conv layer.
+                         Default: None.
+        layer_level_root (list[bool]): dla layer to apply level_root. this
+                                 is only used for tree level.
+        residual_root (bool): whether to use residual in root layer.
+    """
+    arch_settings = {
+        34: (BasicBlock, (1, 1, 1, 2, 2, 1), (16, 32, 64, 128, 256, 512)),
+    }
+
+    def __init__(self,
+                 depth=34,
+                 in_channels=3,
+                 norm_cfg=None,
+                 conv_cfg=None,
+                 layer_level_root=(False, True, True, True),
+                 residual_root=False):
+        super(DLANet, self).__init__()
+        if depth not in self.arch_settings:
+            raise KeyError(f'invalida depth {depth} for DLA')
+        block, levels, channels = self.arch_settings[depth]
+        self.channels = channels
+        self.num_levels = len(levels)
+        self.base_layer = nn.Sequential(
+            build_conv_layer(
+                conv_cfg,
+                in_channels,
+                channels[0],
+                7,
+                stride=1,
+                padding=3,
+                bias=False),
+            dla_build_norm_layer(norm_cfg, channels[0])[1],
+            nn.ReLU(inplace=True))
+
+        for i in range(2):
+            level_layer = self._make_conv_level(
+                channels[0],
+                channels[i],
+                levels[i],
+                norm_cfg,
+                conv_cfg,
+                stride=i + 1)
+            layer_name = f'level{i}'
+            self.add_module(layer_name, level_layer)
+
```

Comment:
Can we add one line comment to indicate that, DLANet first uses two conv then uses several Tree layers?